### ISSUE TITLE
Could not load file or assembly 'System.Net.Http with .NET 4.6.1

### ISSUE THREAD

**Issue by LLRDevGirl** [dotnet/corefx#38495]
System.Net team and @karelz , met you at Build conference and you asked me to submit an issue.  I have 5 webjobs having this problem.  I'm describing only **one** of the five webjob issues here.  **Need your help figuring out how to fix so webjob can be deployed with DevOps Pipeline.** 
Thank you,
Myrna

OrderSync project - .NET 4.6.1., project builds with warnings in Visual Studio.  Our CI/CD pipeline in Azure DevOps shows release is successful.  Yet, webjob throws error on Azure portal.  Have to manually deploy in VS in order to make it to prod successfully.

Method call that errors is ShipOrders that calls FedExBLL and FedExDAL.  Errors related to System.Net.Http versions.  Need your help figuring out how to fix so webjob can be deployed with DevOps Pipeline.

Here's the app insights report on the bug for the webJob:
[query_data (1).xlsx](https://github.com/dotnet/corefx/files/3283585/query_data.1.xlsx)

Here's the references:
![FedEx BLL properties](https://user-images.githubusercontent.com/25438083/59394506-c1c15700-8d34-11e9-9e3e-158de826ea1b.PNG)
![FedExDal properties](https://user-images.githubusercontent.com/25438083/59394507-c1c15700-8d34-11e9-92d0-26eef45f07c2.PNG)
![OrderSync systemnethttp](https://user-images.githubusercontent.com/25438083/59394508-c1c15700-8d34-11e9-96e5-51580892afc3.PNG)

Build output:

[Output-Build for OrderSync.txt](https://github.com/dotnet/corefx/files/3283588/Output-Build.for.OrderSync.txt)

**Comment by karelz** [comment]
First question: Is it possible for you to target (and run on) 4.7.1+? We believe all similar issues are fixed in that release.

@joperezr can you please help take a look?

**Comment by joperezr** [comment]
> First question: Is it possible for you to target (and run on) 4.7.1+? We believe all similar issues are fixed in that release.

Actually, all these issues were fixed in 4.7.2, so it'd be better to target 4.7.2 if possible.

@LLRDevGirl I need a bit more info in order to figure out what is going on. Can you please provide:
 - An msbuild.binlog of your project. In order to produce one, simply run the following on a developer command prompt: `msbuild yourSolution.sln /t:rebuild /bl:msbuild.binlog` which will generate a msbuild.binlog. If you can share that it would be great.
 - Are you able to connect to one of the machines in production with the issue? If so, it would be interesting to know what version of the framework is installed.
 - Are there any binding redirects in your web.config? If so, please paste those.
 - How are you deploying your app to production?
 - What VS are you using? (full version which you can get through Help -> About menu)

**Comment by LLRDevGirl** [comment]
@joperezr Thank you for your assistance. 

- In total I have 4 projects with this problem.  I'll start with OrderSync.  I'm unable to upgrade to 4.7.2 due to dependencies.  However, I'm attaching the msbuild.binlog file.  
- Webjobs are running .NET 4.7.2 (output attached)
- Config redirects attached
- This is in production but I have to deploy from VS 2017 Enterprise v. 15.9.12 (publish as an Azure Webjob).  
- We have our DevOps pipeline which is the preferred method for deployment and I'd like to be able to deploy from here and have the webjob function properly.  

[config redirects.txt](https://github.com/dotnet/corefx/files/3298255/config.redirects.txt)
[version of net framework.txt](https://github.com/dotnet/corefx/files/3298256/version.of.net.framework.txt)

[msbuild.zip](https://github.com/dotnet/corefx/files/3298260/msbuild.zip)

**Comment by joperezr** [comment]
Can you try setting the following two properties in your OrderSync project:

```xml
<PropertyGroup>
  <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
  <DependsOnNETStandard>true</DependsOnNETStandard>
</PropertyGroup>
```

After setting those and rebuilding, you should see a warningin VS telling you that you need to add some binding redirects to your project. Please double click the warning and let VS add these extra binding redirects. Let me know if by doing that this fixes your issue.

**Comment by LLRDevGirl** [comment]
The warning I'm getting after updating the proj file  and setting autoGenerate... to true.  Not sure what else to do.

Found conflicts between different versions of the same dependent assembly. Please set the "AutoGenerateBindingRedirects" property to true in the project file. For more information, see http://go.microsoft.com/fwlink/?LinkId=294190.	OrderSync

**Comment by joperezr** [comment]
Yes that warning is expected. Can you double click it? It should prompt you a dialog asking if you want binding redirects to get added to your web.config. Is that not what you are seeing?

**Comment by ramirez44** [comment]
Nothing happens when I double click it.  I've tried removing the property from the csproj file, rebuilding then re-adding it.  Still nothing.

**Comment by ramirez44** [comment]
btw, this is not a webapp so no web.config; its a app.config file

**Comment by joperezr** [comment]
Oh I see, then setting AutoGenerateBindingRedirects to true would do the trick and should get rid of the warning. Here you can find more info on the different ways on how to set this property to true: https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/how-to-enable-and-disable-automatic-binding-redirection

**Comment by ramirez44** [comment]
I am familiar with that link you sent. I'll double check to make sure I did it correctly.

**Comment by ramirez44** [comment]
Hi Jose, I quadruple checked.  Even though the proj file has the correct AutoGenerateBindingRedirects, for some reason the build is not reading it.  Continues to say this.  I googled and tried a few other options but nothing seems to be working.  Need help.  Thank you.  Should I do the AutoGenerateBindingRedirects to all the referenced projects maybe?

Found conflicts between different versions of the same dependent assembly. **Please set the "AutoGenerateBindingRedirects" property to true in the project file.** For more information, see http://go.microsoft.com/fwlink/?LinkId=294190.	OrderSync

**Comment by joperezr** [comment]
Did you try doing it through the UI? 

1. Right-click the project in Solution Explorer and select Properties.

2. On the Application page, uncheck the Auto-generate binding redirects option.

3. Press Ctrl+S to save the change.

**Comment by joperezr** [comment]
could you perform a binlog and send it our way for us to investigate those warnings? The process of generating a binlog is stated earlier on this issue.

**Comment by LLRDevGirl** [comment]
[msbuild (2).zip](https://github.com/dotnet/corefx/files/3303302/msbuild.2.zip)

Thanks Jose. :)

**Comment by ramirez44** [comment]
@joperezr Any suggestions for me after looking at the binlog?

**Comment by joperezr** [comment]
mm from what I can tell in the binlog, the binding redirects where correctly added to your config file. Did you try opening the file `obj\Debug\OrderSync.exe.config`?  Is there a System.Net.Http binding redirect there forcing version 4.2.0? If so, can you still repro the problem you were having originally? I would expect that when running this app, every assembly that wants a different version of System.Net.Http would just load version 4.2.0.0
